[Task 1469] Add basic rendering tests for boardSummary component

msa2984 opened the pull request:
**Task**:
https://dev.azure.com/csebostoncrew/Retrospectives-AzDO-Extension/_workitems/edit/1469

**Notes**:
- Phong noted on my last PR that the `__mocks__` were not properly formatted so I have reformatted them in this PR. 
- In my investigations to determine if we can add more sophisticated tests beyond basic rendering / snapshot testing, I have discovered that there is a [known issue with the current enzyme adapter](https://github.com/enzymejs/enzyme/issues/2429), as it does not work with React 17 + the `mount` function. The `mount` function will be necessary for verifying when state changes change the state of children that are nested one or two levels within the component, as the `shallow` function only renders basic children of the component. I have updated to the most popular enzyme adapter with React 17 support, and have verified that it resolves the errors from the React 16 adapter. 
- The ` --coverage` auto generates test coverage reports in the coverage folder, with a summary which looks like the below image. We should determine if we want to add the cobertura coverage file to the Github Actions. 

![image](https://user-images.githubusercontent.com/38594207/153299995-e36acf41-06ea-473c-a4bf-fbb295ea6301.png)

msa2984 committed:
:hammer: initial board summary test, fix mocks formatting

msa2984 committed:
:hammer: add test for testing work item counts.

msa2984 committed:
add some tests for displaying work items.

msa2984 committed:
:hammer: intermediary commit

msa2984 committed:
use a different enzyme adapter to resolve issues with mount command.

msa2984 commented on RetrospectiveExtension.Frontend/components/__tests__/mocked_components/WorkItemTracking.tsx:
```diff
@@ -0,0 +1,81 @@
+import { mocked } from 'jest-mock';
```
I'm open to better names for this folder than `mocked_components` ... essentially these components are mocked implementations of classes within the WorkItemTracking class in the auzre api. However since they are not functions / services, we can use these implementations in-line in tests without needing `jest.mock`

msa2984 committed:
:hammer: oops - remove unneeded packages

msa2984 committed:
Merge branch 'dev' into marisaashour/boardsummary-unit-tests

polatengin reviewed (approved):
great, thanks..

phongcao commented:
@msa2984 I really like the amount of details you put in your PR description. Regarding the code coverage, @nexilus18 and I have spent some time trying to add it to our workflow but GitHub Actions doesn't have any built-in support for that and it's not ideal to use any third-party solution.

msa2984 commented:
> @msa2984 I really like the amount of details you put in your PR description. Regarding the code coverage, @nexilus18 and I have spent some time trying to add it to our workflow but GitHub Actions doesn't have any built-in support for that and it's not ideal to use any third-party solution.

Thanks @phongcao !


that's good to know about the code coverage. Do you think I should remove the `--coverage` flag then? I can always add a note in the testing section of the CONTRIBUTING.md that the flag can be added if a contributor is interested in seeing the coverage statistics, I just don't know if it makes sense to run it every build if we don't have a way to view the result.

phongcao commented:
Can you help fix these linter warnings in your PR as well? The last two warnings can be fixed by adding:

```tsx
// eslint-disable-next-line @typescript-eslint/no-explicit-any
} catch (e: any) {
```

<img width="1227" alt="Screen Shot 2022-02-11 at 9 59 30 AM" src="https://user-images.githubusercontent.com/9741323/153615042-1b045b82-c62a-4e73-bf22-d3f6f626e467.png">

phongcao commented:
> > @msa2984 I really like the amount of details you put in your PR description. Regarding the code coverage, @nexilus18 and I have spent some time trying to add it to our workflow but GitHub Actions doesn't have any built-in support for that and it's not ideal to use any third-party solution.
> 
> Thanks @phongcao !
> 
> that's good to know about the code coverage. Do you think I should remove the `--coverage` flag then? I can always add a note in the testing section of the CONTRIBUTING.md that the flag can be added if a contributor is interested in seeing the coverage statistics, I just don't know if it makes sense to run it every build if we don't have a way to view the result.

Yeah, I think we should do that and it wouldn't increase the current runtime of our workflow.

msa2984 committed:
Some linting resolutions - also, add --coverage documentation to the
CONTRIBUTING doc and removed test that is skipped.

msa2984 commented:
okay I have fixed the code to remove the eslint warnings - there were two warnings about unexpected `any` in the `itemDataService.tsx` and `dataService.tsx` files, and although I did not change those files in this review, I have updated them to disable the any warning as well. 

I have also added a note to the CONTRIBUTING.md to describe generating test coverage reports.

msa2984 committed:
Merge branch 'dev' into marisaashour/boardsummary-unit-tests

phongcao commented:
> okay I have fixed the code to remove the eslint warnings - there were two warnings about unexpected `any` in the `itemDataService.tsx` and `dataService.tsx` files, and although I did not change those files in this review, I have updated them to disable the any warning as well.
> 
> I have also added a note to the CONTRIBUTING.md to describe generating test coverage reports.

Well done, thanks!
